### DISCUSSION TITLE
Got a temporarily redirect response 307 when hosting fastAPI in AWS

### DISCUSSION THREAD

**Discussion by minhdc** [fastapi/fastapi#8528] [Questions]
### First Check

- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
my_api_prefix = /authz/v5
app = FastAPI(
    swagger_ui_parameters={"docExpansion": "none"},
    title="My Swagger",
    docs_url=my_api_prefix + "/docs"
)
origins = ["*"]

app.add_middleware(
    CORSMiddleware,
    allow_origins=origins,
    allow_credentials=True,
    allow_methods=["*"],
    allow_headers=["*"],
)


app.add_event_handler("startup", create_start_app_handler(app))
app.add_event_handler("shutdown", create_stop_app_handler(app))


if __name__ == "__main__":
    uvicorn.run(app, host="0.0.0.0", port=8001)
```

### Description

It's okay to access swagger UI when I run my app locally. However, when I hosted my app in an AWS instance, I cannot access swagger UI. My logged response when using curl from my laptop is:

```
> GET /authz/v5/docs HTTP/1.1
> Host: api-dev.something.com
> User-Agent: curl/7.71.1
> Accept: */*
> 
* Mark bundle as not supporting multiuse
< HTTP/1.1 307 Temporary Redirect
< Date: Thu, 21 Jun 2022 03:15:36 GMT
< Content-Type: application/json
< Content-Length: 0
< Connection: keep-alive
< x-amzn-RequestId: b4877b45-ac90-484b-9a2b-3d2d4fbcfb8f
< x-amzn-Remapped-content-length: 0
< x-amz-apigw-id: Un8h2Fn3yWoFdag=
< x-amzn-Remapped-server: uvicorn
< location: http://webse-WebLo-some-messy-stuff-here.elb.ap-southeast-1.amazonaws.com/authz/v5/docs 
< x-amzn-Remapped-date: Thu, 21 Jun 2022 03:15:36 GMT
< 
* Connection #0 to host api-dev.something.comleft intact

``` 

### Operating System

Linux

### FastAPI Version

0.75.0

### Python Version

3.8.8

### Additional Context
Updated: 
I tried the following config, at least I was able to get to the swagger page but it couldn't found the **openapi.json** . I turn on the developer mode on my browser (Firefox/Ubuntu) and see a 404 in the request for **openapi.json**
```
app = FastAPI(
    docs_url=my_api_prefix + "/docs" + "/",    
    openapi_url=my_api_prefix + "/openapi.json", 
)
```

Here is the error message:
```
Fetch error
NetworkError when attempting to fetch resource. /authz/v1/openapi.json
```

P/s: 
- I encountered the same issue when hosting Django on AWS but I changed the **base_url** of **drf-yasg** config and it worked just fine. However, in FastAPI, I couldn't find the equivalent configuration.
- I ssh to my AWS instance and:``` curl -v GET http://localhost:my_port/my_api_prefix/openapi.json``` then I was able to get the json definition normally, so I guess this related to network config or some flags inside FastAPI
- I tried to send test requests from the API gateway but still got the same error

_No response_

**Comment by minhdc** [comment]
I tried to edit **root_path** according to this doc: 
https://fastapi.tiangolo.com/advanced/behind-a-proxy/#checking-the-current-root_path
but it doesn't work anyway.

**Comment by JarroVGIT** [comment]
In what kind of service are you running this app? I see it is redirecting to an http endpoint (rather than a https) so there might be an issue with that? You could try adding the httpsredirect middleware to your application, but that is just guessing on my part to be honest..

**Comment by minhdc** [comment]
> In what kind of service are you running this app? I see it is redirecting to an http endpoint (rather than a https) so there might be an issue with that? You could try adding the httpsredirect middleware to your application, but that is just guessing on my part to be honest..

I'm  building an authorization service based on casbin lib. 
Yeah, I see this too:
```
< location: http://webse-WebLo-some-messy-stuff-here.elb.ap-southeast-1.amazonaws.com/authz/v5/docs 
```
I still don't know why its redirecting to this anw, 
Yeah, maybe I need to add an appropriate middleware to redirect it into a correct address.

Updated:
I tried the new configuration (edited above). It showed the swagger page but couldn't found the **openapi.json**

**Comment by prabal01pathak** [comment]
#4211

**Comment by minhdc** [comment]
> #4211

Thanks, but when I follow the suggested solution, I encountered another 403 error:
![image](https://user-images.githubusercontent.com/25708113/175853921-4f776c5a-bc77-4fac-910d-8aaaf3c1bdc6.png)

**Comment by kyrylo-khatsko-MDTU-VOL** [comment]
I have internal redirect in AWS lambda+mangum+fastAPI
neither with or without CORS

But locally everything is working fine.

I did a lot of debugging. 307 comes from fast API, not Mangum. To exactly the same URL, it is asking to.

**Comment by claritytech1008** [comment]
+1 looking to solve this

**Comment by jianzs** [comment]
Any progress on this?

**Comment by nosachamos** [comment]
I see the same issue. Configuring a reverse proxy on fast api to check for x-forwarded-proto and setting preferred scheme to http (which is the value sent by the ALB on the proto header) does not help. This solution worked for Flask, but Fast API seems to not take the scheme into account.
